docs/configuration/config-file-reference.md

```diff
@@ -2822,6 +2827,14 @@ The `limits_config` configures default and per-tenant limits imposed by Cortex s
 # CLI flag: -frontend.max-cache-freshness
 [max_cache_freshness: <duration> | default = 1m]
 
+# Maximum number of queriers that can handle requests for single user. If set to
+# 0, or value less than number of available queriers, all queriers will handle
+# given user's requests. Each frontend will select the same queriers (if they
+# are connected to all frontends). Note that this only works with queriers
+# connecting to the query-frontend, not when using downstream URL.
+# CLI flag: -frontend.max-queriers-per-user
+[max_queriers_per_user: <int> | default = 0]
```

Comment:
Store-gateways can use different sharding strategies. In case of queriers, no sharding was done before (which corresponds to using `max_queriers_per_user: 0`), and only shuffle-sharding is available (if `max_queriers_per_user` is greater than 0). We can introduce sharding_strategy option too, but personally I don't see a need for it here.

Reply:
I agree it's not required here, but could help with config consistency.

Reply:
> Food for thought (not having a strong opinion): I agree it's not required here, but could help with config consistency.

If we want to be consistent with other config options, we should also support "default" (non-shuffle-sharding) strategy with non-zero shard size.

Reply:
Let's keep it as is for now. We marked this feature as experimental, which will allow us to eventually fine-tune the config before marking it stable.